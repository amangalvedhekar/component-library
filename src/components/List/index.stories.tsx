import * as React from 'react';
import {checkA11y} from '@storybook/addon-a11y';
import {withInfo} from '@storybook/addon-info';
import {storiesOf} from '@storybook/react';
import {borderType, List, StyledUl} from "./";
import {Link} from "../Link";
import CheckBox from "../Forms/CheckBox/";
import {boolean} from "@storybook/addon-knobs";

const data = ["dividends", "fees", "trade", "deposits"];
const packItems = [
    {
        isPacked: false,
        name: "Abc",
    },
    {
        isPacked: false,
        name: "Def",
    },
    {
        isPacked: false,
        name: "Ghi",
    },
    {
        isPacked: false,
        name: "Jkl",
    }
];

storiesOf("List", module)
    .addDecorator(checkA11y)
    .addDecorator(withInfo)
    .add("with tab index and all border", () => (
        <StyledUl>
            <List data={"abc"} tabIndex={0} borderType={borderType.all}>
                <Link href={"/something"}>
                    {"abc"}
                </Link>
            </List>
        </StyledUl>
    ))
    .add("with out tab index and bottom border", () => (
        <StyledUl>
            <List data={"abc"} borderType={borderType.bottom}>
                <Link href={"/something"}>
                    {"abc"}
                </Link>
            </List>
        </StyledUl>
    ))
    .add("with out tab index and top border", () => (
        <StyledUl>
            <List data={"abc"} borderType={borderType.top}>
                <Link href={"/something"}>
                    {"abc"}
                </Link>
            </List>
        </StyledUl>
    ))
    .add("List of items", () => (<StyledUl>
        {data.map(elm => (
            <List data={elm} borderType={borderType.bottom} tabIndex={0} key={elm}>
                <Link href={"#"}>
                    {elm}
                </Link>
            </List>
        ))}
    </StyledUl>))
    .add("List with checkbox and link", () => (<StyledUl>
        {packItems.map(elm => (
            <List
                data={elm.name}
                borderType={borderType.none}
                tabIndex={0}
                key={elm.name}
            >
                <CheckBox
                    name={elm.name}
                    labelname={elm.name}
                    checked={boolean("Is checked", false)}
                />
                <Link href={"#"}>
                    {"Remove"}
                </Link>
            </List>))}
    </StyledUl>));
