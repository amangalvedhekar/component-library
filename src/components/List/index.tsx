import React from "react";
import styled from 'styled-components';


export enum borderType {
    all = "all",
    bottom = "bottom",
    top = "top",
    none = "none"
}

export interface IList {
    children: React.ReactNode;
    data: string;
    tabIndex?: number;
    borderType: borderType;
    onClick?: (e: React.SyntheticEvent) => void;
    onKeyDown?: (e: any) => void;
}

export const StyledUl = styled("ul")`
    list-style: none;
`;
export const getBorderForListItem = (border : borderType | undefined) => {
    if (border === undefined) {
        border = borderType.none;
    }
    return {
        top: "border-top: 1px solid grey",
        bottom: "border-bottom: 1px solid grey",
        all: "border: 1px solid grey",
        none: "border: none"
    }[border];
};

const StyledLI: any = styled("li")`
    margin: 8px;
    ${(props: IList) => getBorderForListItem(props.borderType)};
    padding: 8px;
`;


export const List = ({children, data, borderType, ...props}: IList) => (<StyledLI
    borderType={borderType}
    {...props}
>
    {children}
</StyledLI>);

List.defaultProps = {
    borderType: "all"
};