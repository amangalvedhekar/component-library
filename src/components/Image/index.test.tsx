import React from 'react';
import {shallow} from 'enzyme';
import StyledImage from "./";
import {borderType} from "../List";

describe("Image component", () => {
    it("is rendered", () => {
        const wrapper = shallow(<StyledImage
            src={"https://placeimg.com/640/480/any"}
            alt={"some text"}
            borderType={borderType.all}
        />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render with default border type", () => {
        const wrapper = shallow(<StyledImage
            src={"https://placeimg.com/640/480/any"}
            alt={"some text"}
        />);
        expect(wrapper.props())
            .toHaveProperty("borderType", "none");
    });
});