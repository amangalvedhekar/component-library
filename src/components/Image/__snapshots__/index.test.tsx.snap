// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component is rendered 1`] = `
<styled.img
  alt="some text"
  borderType="all"
  src="https://placeimg.com/640/480/any"
/>
`;
